clientd3d/preferences.c

```diff
@@ -0,0 +1,1718 @@
+        case IDC_WITHDRAW:
+            AssignKey(hDlg, TCWithdraw, IDC_WITHDRAW);
+            break;
+        case IDC_ATTACK_MOD:
+            ModifyKey(hDlg, TCAttack, IDC_ATTACK);
+            break;
+        case IDC_BUY_MOD:
+            ModifyKey(hDlg, TCBuy, IDC_BUY);
+            break;
+        case IDC_DEPOSIT_MOD:
+            ModifyKey(hDlg, TCDeposit, IDC_DEPOSIT);
+            break;
+        case IDC_EXAMINE_MOD:
+            ModifyKey(hDlg, TCExamine, IDC_EXAMINE);
+            break;
+        case IDC_LOOK_MOD:
+            ModifyKey(hDlg, TCLook, IDC_LOOK);
+            break;
+        case IDC_OFFER_MOD:
+            ModifyKey(hDlg, TCOffer, IDC_OFFER);
+            break;
+        case IDC_OPEN_MOD:
+            ModifyKey(hDlg, TCOpen, IDC_OPEN);
+            break;
+        case IDC_PICKUP_MOD:
+            ModifyKey(hDlg, TCPickup, IDC_PICKUP);
+            break;
+        case IDC_WITHDRAW_MOD:
+            ModifyKey(hDlg, TCWithdraw, IDC_WITHDRAW);
+            break;
+        }
+        break;
+
+    case WM_NOTIFY:
+        if (((LPNMHDR)lParam)->code == PSN_APPLY) {
+            // Save settings
+            UpdateINIFile();
+            SetWindowLongPtr(hDlg, DWLP_MSGRESULT, PSNRET_NOERROR);
+            return (INT_PTR)TRUE;
+        } else if (((LPNMHDR)lParam)->code == PSN_RESET) {
+            // Handle Cancel button
+            SetWindowLongPtr(hDlg, DWLP_MSGRESULT, PSNRET_NOERROR);
+            return (INT_PTR)TRUE;
+        }
+        break;
+    }
+    return (INT_PTR)FALSE;
+}
+
+// Implementation of OptionsPreferencesDlgProc
+INT_PTR CALLBACK OptionsPreferencesDlgProc(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    switch (message)
+    {
+    case WM_INITDIALOG:
+    case WM_USER_REINITDIALOG:
+        // Initialize dialog with current settings
+        CheckDlgButton(hDlg, IDC_ALWAYSRUN, m_alwaysrun);
+        CheckDlgButton(hDlg, IDC_CLASSIC, m_classic);
+        CheckDlgButton(hDlg, IDC_DYNAMIC, m_dynamic);
+        CheckDlgButton(hDlg, IDC_QUICKCHAT, m_quickchat);
+        CheckDlgButton(hDlg, IDC_SOFTWARE, m_software);
+        CheckDlgButton(hDlg, IDC_ATTACKONTARGET, m_attackontarget);
+        CheckDlgButton(hDlg, IDC_GPU_EFFICIENCY, m_gpuefficiency);
+        return (INT_PTR)TRUE;
+    case WM_COMMAND:
+        switch (LOWORD(wParam))
+        {
+        case IDC_CLASSIC:
+            m_classic = IsDlgButtonChecked(hDlg, IDC_CLASSIC);
+            if (m_classic)
+            {
+                MessageBox(hDlg, "Activating \"Classic Key Bindings\" will cause the game to ignore the key bindings!", "Notice", MB_OK);
+            }
+            break;
+
+        case IDC_SOFTWARE:
+            m_software = IsDlgButtonChecked(hDlg, IDC_SOFTWARE);
+            if (!m_software)
+            {
+                MessageBox(hDlg, "Software renderer may have been activated due to a problem with your video setup.\n\t\t Disabling may cause the game not to run.", "Warning", MB_OK);
```

Comment:
I guess these strings were untranslated in keybind?  We should move them to the resource file so that they can be translated in the future.